### PR TITLE
feat: add ListSum transformer and layer

### PR DESCRIPTION
**Pull request by dummy1cx**
### Description

Adds `ListSumTransformer` (Spark) and `ListSumLayer` (Keras), extending the listwise
aggregation family alongside `ListMax`. The implementation mirrors `ListMax` in
structure, parameters and naming:

- `topN` + `sortOrder` — sum over the top N items only, to approximate production
  statistics where the full list is not available at serving time.
- `withSegment` — sum within segments of a second column rather than across the whole
  query. For example, the summed price of hotels within each star rating in a query.
- `minFilterValue` — exclude padded values before aggregating.

Motivation: multi-level ranking models need listwise sums aggregated per segment, which
currently has no equivalent in the library.

### Keras Layer Checklist
Verify that:
- [x] The new Keras layer extends BaseLayer
- [x] The `_call` method has been implemented in the new layer.
- [x] The `compatible_dtypes` property is defined in the new layer.
- [x] The new layer is decorated with `@tf.keras.utils.register_keras_serializable(package=kamae.__name__)`.
- [x] The new layer takes a `name`, `input_dtype`, and `output_dtype` as arguments to the constructor and that this is passed to the super constructor.
- [x] The Keras layer is serializable. I have implemented the `get_config` method.
- [x] There are unit tests of the new layer.
- [x] There is a specific test of layer serialisation added.
- [x] The new layer is imported in the `__init__.py` file in the `layers` directory.

### Spark Transformer/Estimator Checklist
Verify that:
- [x] The new Spark Transformer extends BaseTransformer.
- [x] If the new transform needs a fit method, a Spark Estimator has been implemented that extends BaseEstimator. (N/A — no fit required.)
- [x] The instructions in the above docs page have been followed for the `__init__` and `setParams` methods.
- [x] The transformer uses one of the input/output mixin classes from base.py.
- [x] If the new transformer requires more parameters that would need to be serialised to the Spark ML pipeline, there is an implemented parameter class. (Reuses the existing `ListwiseStatisticsParams` and `NanFillValueParams`.)
- [x] The `compatible_dtypes` property has been implemented to specify the input/output data types that my transformer/estimator supports.
- [x] A Keras subclassed layer is returned in the transformer's `get_tf_layer` method.
- [x] There are unit tests of the new transform. In particular, there are parity tests between the Spark and Keras implementations.
- [x] The new transformer/estimator is imported in the `__init__.py` file in the `transformers` directory.

Finally, please verify that:
- [x] There is a new entry (alphabetical order) in the README table describing the new layer/transformer

### Notes for reviewers

Three decisions I would like a second opinion on:

1. **`minFilterValue` semantics.** Values below the threshold contribute `0` to the sum
   rather than being dropped from the segment. The masking uses `tf.where` with
   `tf.zeros_like` rather than a NaN-based approach, deliberately avoiding float-only ops
   so that integer value columns work. Happy to change if you would prefer different
   semantics here.

2. **`string` in `compatible_dtypes`.** This is not to sum strings. `BaseLayer`
   validates the dtype of *every* input tensor, and under `withSegment=True` the second
   input is a segment identifier which is commonly a string. Without `string` in the
   list, string segment keys are rejected.

3. **Known parity gap, deliberately not addressed here.** If a segment becomes empty
   after `minFilterValue` filtering, the Spark side applies `nanFillValue` via `fillna`
   while the Keras layer returns `0.0`. Reaching this requires both `minFilterValue` and
   a non-zero `nanFillValue`, which no current configuration sets. Closing it properly
   needs `map_fn_w_axis` to accept more than two tensors, so I have kept it out of this
   PR to keep the change reviewable. Happy to fold it in here instead if you would rather
   not merge a known gap.

### Testing

Unit tests for both implementations cover the base case, `minFilterValue`, `topN` with
and without a filter, `topN` greater than the list size, segmentation, segmentation with
multiple features, segmentation with string segment IDs, and segmentation combined with
`minFilterValue`, plus the `ValueError` paths. The Spark tests include Spark/TensorFlow
parity cases. The layer is also registered in the JIT-compatibility and serialisation
meta-test suites.

### CODE REVIEW

File: src/kamae/keras/tensorflow/layers/list_sum.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+        - If with_segment = False: the layer calculates the sum of the first tensor
+    based on second tensor's topN items in the same given axis dimension.
+    By using the topN items to calculate the statistics, we can better approximate
+    the real statistics in production. It is suggested to use a large enough topN to
+    get a good approximation of the statistics, and an important feature to sort on,
+    such as item's past production.
+
+    Example: calculate the sum price in the same query, based only on the top N
+    items sorted by descending production.
```
Comment (jamesdshinner) [inline src/kamae/keras/tensorflow/layers/list_sum.py:50]: Let's ensure consistent indentation here

Comment (jamesdshinner) [reply src/kamae/keras/tensorflow/layers/list_sum.py:50]: /I think you have two separate paragraphs flowing together

**Review by georyetti — DISMISSED** [review]
LGTM

File: src/kamae/spark/transformers/list_sum.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+
+class ListSumTransformer(
+    BaseTransformer,
+    SingleInputSingleOutputParams,
+    MultiInputSingleOutputParams,
+    ListwiseStatisticsParams,
+    NanFillValueParams,
+):
+    """
```
Comment (jamesdshinner) [inline src/kamae/spark/transformers/list_sum.py:52]: Same point on indentation and formatting here

**Review by jamesdshinner — CHANGES_REQUESTED** [review]
Couple of pedantic style comments only

File: src/kamae/keras/tensorflow/layers/list_sum.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        return [
+            "bfloat16",
+            "float16",
+            "float32",
+            "float64",
+            "int8",
+            "int16",
+            "int32",
+            "int64",
+            "string",
```
Comment (mandrecki) [inline src/kamae/keras/tensorflow/layers/list_sum.py]: What is the expected behaviour for string type?
if this is intended, then add tests for this case and state expected behaviour in docstring.

File: src/kamae/spark/transformers/list_sum.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        withSegment: bool = False,
+        minFilterValue: Optional[float] = None,
+        nanFillValue: float = 0.0,
+    ) -> None:
+        super().__init__()
+        self._setDefault(
+            topN=None,
+            sortOrder="asc",
+            minFilterValue=None,
+            nanFillValue=0,
```
Comment (mandrecki) [inline src/kamae/spark/transformers/list_sum.py]: should be
```
nanFillValue=nanFillValue,
```

File: src/kamae/keras/tensorflow/layers/list_sum.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        :param nan_fill_value: The value to fill NaNs results with. Defaults to 0.
+        :param axis: The axis to calculate the statistics across. Defaults to 1.
+        """
+        super().__init__(
+            name=name, input_dtype=input_dtype, output_dtype=output_dtype, **kwargs
+        )
+        self.top_n = top_n
+        self.sort_order = sort_order
+        self.min_filter_value = min_filter_value
+        self.nan_fill_value = nan_fill_value
```
Comment (mandrecki) [inline src/kamae/keras/tensorflow/layers/list_sum.py]: is this used anywhere? let's add a test in parity check with non-default nan fill value

Comment (mandrecki) [inline src/kamae/keras/tensorflow/layers/list_sum.py:96]: is this used anywhere? let's add a test in parity check that with a non-default `nan_fill_value`

**Review by mandrecki — CHANGES_REQUESTED** [review]
